### ISSUE TITLE
Containers failing to start with /dev/pts mounts

### ISSUE THREAD

**Issue by vijaylbais** [lxc/lxc#1054]
Hello guys,

I've been having an issue with LXC lately where it refuses to start containers with an error pointing to /dev/pts/<pts no>. Any pointers will be greatly appreciated. My setup is as mentioned. 

kernel version: 3.10.101-1.el6.elrepo.x86_64
lxc version: 1.0.8-1
os: centos 6.8
LXC path: /var/lib/lxc
Max ptys: 
$ cat /proc/sys/kernel/pty/max 
4096

**ERROR:**
lxc-start: utils.c: open_without_symlink: 1317 Input/output error - Error examining console in /usr/lib64/lxc/rootfs/dev/console
lxc-start: conf.c: setup_dev_console: 1844 failed to mount '/dev/pts/30' on '/usr/lib64/lxc/rootfs/dev/console'
lxc-start: conf.c: lxc_setup: 4316 failed to setup the console for '<container name>'
lxc-start: start.c: do_start: 703 failed to setup the container
lxc-start: sync.c: __sync_wait: 51 invalid sequence number 1. expected 2
lxc-start: start.c: __lxc_start: 1100 failed to spawn '<container name>'
lxc-start: lxc_start.c: main: 341 The container failed to start.
lxc-start: lxc_start.c: main: 345 Additional information can be obtained by setting the --logfile and --logpriority options.

**Comment by rumblpak** [comment]
Check your mount point for /dev/pts. Its probably mounted incorrectly. This seems to happen to me every so often while starting and stopping containers too.

**Comment by slavekjurkowski2** [comment]
I'm seeing this same issue. A work around for me is remounting /dev/pts in read/write as it seems to have gone read only but i can't find a root cause. 

`mount -o remount,rw /dev/pts
`
I'm on centos 6.8. Have you found a root cause?

**Comment by brauner** [comment]
This is `LXC 1.0.8`? This is not even the latest `1.0.*` stable release. If you can, then you should update to latest `LXC` and try to reproduce. There'll be another `2.0.*` stable release soon which will container a bunch of console and pts device handling improvements I've pushed.

**Comment by brauner** [comment]
This issue seems to have gone stale. I've also rewritten most of that code in the meantime so this is likely fixed. If this is still an issue, please re-open.

**Comment by dceee** [comment]
I am currently running into the same issue:
`[root@lxc1 ~]# lxc-start -n centos3
lxc-start: centos3: lxccontainer.c: wait_on_daemonized_start: 855 No such file or directory - Failed to receive the container state
lxc-start: centos3: tools/lxc_start.c: main: 330 The container failed to start
lxc-start: centos3: tools/lxc_start.c: main: 333 To get more details, run the container in foreground mode
lxc-start: centos3: tools/lxc_start.c: main: 336 Additional information can be obtained by setting the --logfile and --logpriority options
[root@lxc1 ~]# mount /dev/pts -o remount,rw
[root@lxc1 ~]# lxc-start -n centos3
[root@lxc1 ~]# 
[root@lxc1 ~]# mount | grep pts
devpts on /dev/pts type devpts (rw,nosuid,noexec,relatime,seclabel,mode=600,ptmxmode=000)
[root@lxc1 ~]# lxc-stop -n centos3
[root@lxc1 ~]# mount | grep pts
devpts on /dev/pts type devpts (ro,nosuid,noexec,relatime,seclabel,mode=600,ptmxmode=000)
[root@lxc1 ~]# 
`
My system:
`
[root@lxc1 ~]# uname -r
3.10.0-957.5.1.el7.x86_64
[root@lxc1 ~]# lxc-start --version
3.1.0
[root@lxc1 ~]# cat /etc/redhat-release 
CentOS Linux release 7.6.1810 (Core) 
[root@lxc1 ~]# rpm -qa 'lxc*'
lxc-3.1.0-0.2.el7.x86_64
lxc-templates-3.1.0-0.2.el7.x86_64
lxc-templates-extra-3.0.3-0.2.el7.noarch
lxc-libs-3.1.0-0.2.el7.x86_64
`

**Comment by brauner** [comment]
What does `dmesg` show?

**Comment by dceee** [comment]
Couldn't spot anything suspicious in dmesg. Looks like the problem is related with my current setup:
I need to keep my containers on the network. Since aufs is not available anymore and overlay is not working on NFS shares, I created an xfs (also tried ext4) formatted image and mounted it to /var/lib/lxc. Now I can use overlay but I am also getting this error.

**Comment by brauner** [comment]
Hm, it might be that you're not allowed to mount another instance of `devpts` in another mount namespace as `rw` if it's mounted as `ro` in the host mount namespace...

**Comment by dceee** [comment]
For some reason some of my freshly installed Centos containers caused this issue while others didn't. So I tried deploying the machines in multiple ways:
Going the download way by `lxc-create -n mycentos -t download` reproducibly caused the problem after a while once the machine has been shut down. 
Deploying machines by `lxc-create -n mycentos -t centos -- -R 7 -a x86_64` never caused any issues. I have no idea if there is any difference between those methods or if I am missing something. I just sticked to the ladder method and never had any problems with /dev/pts.
